python/lib/core/dmod/core/common/collections/bag.py

```diff
@@ -0,0 +1,126 @@
+"""
+Defines a collection of unordered data not requiring a hash
+"""
+from __future__ import annotations
+
+import typing
+
+from typing_extensions import Self
+
+
+_T = typing.TypeVar("_T")
+
+
+class Bag(typing.Collection[_T]):
```

Comment:
It lets you do set comparisons with collections of non-hashable items.

So:

```python
[unhash1, unhash2, unhash3, unhash2] == [unhash1, unhash2, unhash3, unhash2]
[unhash1, unhash2, unhash3, unhash2] != [unhash1, unhash2, unhash3]
[unhash1, unhash2, unhash3] != [unhash2, unhash3, unhash1]
{unhash1, unhash2, unhash3} != {unhash2, unhash3, unhash1}
# Throws an error because the unhash objects can't be hashed and therefore can't be in a set

Bag([unhash1, unhash2, unhash3]) == Bag([unhash3, unhash2, unhash1])
Bag([unhash1, unhash2, unhash3]) == Bag([unhash3, unhash2, unhash1, unhash2])
```

Reply:
Maybe I'm forgetting something subtle about the other dunders, but doesn't this need an `__eq__` implementation for that?  Or does this inherit an implementation (beyond identity comparison) by extending  _Collection_?